tests/unit/test_unique_any.cpp

```diff
@@ -0,0 +1,394 @@
+        auto p = util::any_cast<int>(&a);
+        static_assert(std::is_same<const int*, decltype(p)>::value,
+                "any_cast(const any*) should return const*");
+    }
+}
+
+// test any_cast(unique_any&)
+TEST(unique_any, any_cast_ref) {
+    util::unique_any ai(42);
+    auto& i = util::any_cast<int&>(ai);
+
+    EXPECT_EQ(typeid(i), typeid(int));
+    EXPECT_EQ(i, 42);
+
+    // any_cast<T>(unique_any&) returns a 
+    i = 100;
+    EXPECT_EQ(util::any_cast<int>(ai), 100);
+}
+
+// test any_cast(const unique_any&)
+TEST(unique_any, any_cast_const_ref) {
+    const util::unique_any ai(42);
+    auto& i = util::any_cast<const int&>(ai);
+
+    EXPECT_EQ(typeid(i), typeid(int));
+    EXPECT_EQ(i, 42);
+
+    EXPECT_TRUE((std::is_same<const int&, decltype(i)>::value));
+}
+
+// test any_cast(any&&)
+TEST(unique_any, any_cast_rvalue) {
+    auto moved = util::any_cast<moveable>(util::unique_any(moveable()));
+    EXPECT_EQ(moved.moves, 2);
+    EXPECT_EQ(moved.copies, 0);
+}
+
+TEST(unique_any, std_swap) {
+    util::unique_any a1(42);
+    util::unique_any a2(3.14);
+
+    auto pi = util::any_cast<int>(&a1);
+    auto pd = util::any_cast<double>(&a2);
+
+    std::swap(a1, a2);
+
+    // test that values were swapped
+    EXPECT_EQ(util::any_cast<int>(a2), 42);
+    EXPECT_EQ(util::any_cast<double>(a1), 3.14);
+
+    // test that underlying pointers did not change
+    EXPECT_EQ(pi, util::any_cast<int>(&a2));
+    EXPECT_EQ(pd, util::any_cast<double>(&a1));
+}
+
+// test operator=(any&&)
+TEST(unique_any, assignment_from_rvalue) {
+    using util::unique_any;
+    using std::string;
+
+    auto str1 = string("one");
+    auto str2 = string("two");
+    unique_any a(str1);
+
+    unique_any b;
+    b = std::move(a); // move assignment
+
+    EXPECT_EQ(str1, util::any_cast<string>(b));
+
+    EXPECT_EQ(nullptr, util::any_cast<string>(&a));
+}
+
+// test template<typename T> operator=(T&&)
+TEST(unique_any, assignment_from_value) {
+    using util::unique_any;
+    std::vector<int> tmp{1, 2, 3};
+
+    // take a pointer to the orignal data to later verify
+    // that the value was moved, and not copied.
+    auto ptr = tmp.data();
+
+    unique_any a;
+    a = std::move(tmp);
+
+    auto vec = util::any_cast<std::vector<int>>(&a);
+
+    // ensure the value was moved
+    EXPECT_EQ(ptr, vec->data());
+
+    // ensure that the contents of the vector are unchanged
+    std::vector<int> ref{1, 2, 3};
+    EXPECT_EQ(ref, *vec);
+}
+
+TEST(unique_any, make_unique_any) {
+    using util::make_unique_any;
+    using util::any_cast;
+
+    {
+        auto a = make_unique_any<int>(42);
+
+        EXPECT_EQ(typeid(int), a.type());
+        EXPECT_EQ(42, any_cast<int>(a));
+    }
+
+    // check casting
+    {
+        auto a = make_unique_any<double>(42u);
+
+        EXPECT_EQ(typeid(double), a.type());
+        EXPECT_EQ(42.0, any_cast<double>(a));
+    }
+
+    // check forwarding of parameters to constructor
+    {
```

Comment:
For thoroughness, checking a 0-parameter and 2- (or more) parameter constructor would help ensure that the correct variadic implementation of `make_unique_any` was deployed. (Also applies to `make_any`).